papas/aliceproto/Identifier.py

```diff
@@ -0,0 +1,50 @@
+# simplified class to provide a unique identifier for each object
+# could also add more information into the identifier as needed
+class Identifier(long) :
+    class PFOBJECTTYPE:
+        NONE=0
+        ECALCLUSTER=1 #simplistic initial implementation (may need an enum for layer to be added)
+        HCALCLUSTER=2
+        TRACK=3
+        PARTICLE=4
+        RECPARTICLE=5
+        BLOCK=6
+    
+    @staticmethod    
+    def makeID(item, type) :
+        x=id(item)
+        value=  type <<40
+        return value | x
+   
+    @staticmethod      
+    def getUniqueID( ident) :
+        return ident & 0b1111111111111111111111111111111111111111
+    
+    @staticmethod  
+    def gettype ( ident) :
+        return ident >> 40
+    
+    @staticmethod  
+    def isECAL ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.ECALCLUSTER  
+
+    @staticmethod  
+    def isHCAL ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.HCALCLUSTER  
+
+
+    @staticmethod  
+    def isTrack ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.TRACK 
+    
+    @staticmethod  
+    def isBlock ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.BLOCK     
+    
+    @staticmethod  
+    def isRecParticle ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.RECPARTICLE 
+    
+    @staticmethod  
+    def isParticle ( ident) :
+        return Identifier.gettype(ident)  ==Identifier.PFOBJECTTYPE.PARTICLE     
```

Comment:
nice identifier class! 
just one thing: please always think about adding a new line at the end of files.